src/object_pool.hpp

```diff
@@ -0,0 +1,226 @@
+ *
+ * There is some memory overhead. Because empty blocks of data are used as borders between allocations.
+ * That way the deallocation algorithm knows where to stop deallocation. That means a ObjectPool<char, 256>
+ * holds a little bit less than 256 characters.
+ *
+ * @author  Julian van Doorn
+ *
+ * @code{.cpp}
+ObjectPool<char, 64> pool = { };
+std::cout << pool << std::endl;                             // ObjectPool with availabilityBitmask:
+1111111111111111111111111111111111111111111111111111111111111111
+char* p0 = pool.allocateBlocks(13);
+std::cout << "Allocated 13 bytes" << std::endl;             // Allocated 13 bytes
+std::cout << pool << std::endl;                             // ObjectPool with availabilityBitmask:
+0000000000000111111111111111111111111111111111111111111111111111
+p0[0] = 'H';
+p0[1] = 'e';
+p0[2] = 'l';
+p0[3] = 'l';
+p0[4] = 'o';
+p0[5] = ' ';
+p0[6] = 'w';
+p0[7] = 'o';
+p0[8] = 'r';
+p0[9] = 'l';
+p0[10] = 'd';
+p0[11] = '!';
+p0[12] = '\0';
+char* p1 = pool.allocateBlocks(13);
+std::cout << "Allocated another 13 bytes" << std::endl;     // Allocated another 13 bytes
+std::cout << pool << std::endl;                             // ObjectPool with availabilityBitmask:
+0000000000000100000000000001111111111111111111111111111111111111
+strcpy(p1, p0);
+p0[6] = 'R';
+p0[7] = '2';
+p0[8] = 'D';
+p0[9] = '2';
+p0[10] = '\0';
+std::cout << p0 << std::endl;                               // Hello R2D2
+std::cout << p1 << std::endl;                               // Hello world!
+pool.deallocateBlocks(p0);
+std::cout << "Deallocated 13 b  ytes (p0)" << std::endl;      // Deallocated 13 bytes (p0)
+std::cout << pool << std::endl;                             // ObjectPool with availabilityBitmask:
+1111111111111100000000000001111111111111111111111111111111111111
+uint32_t* myInt = (uint32_t*) pool.allocateBlocks(4);
+std::cout << "Allocated 4 bytes" << std::endl;              // Allocated 4 bytes
+std::cout << pool << std::endl;                             // ObjectPool with availabilityBitmask:
+0000111111111100000000000001111111111111111111111111111111111111
+*myInt = 0xDEADBEEF;
+std::cout << std::hex << *myInt << std::endl;               // deadbeef
+pool.deallocateBlocks(p1);
+pool.deallocateBlocks((char*) myInt);
+std::cout << "Deallocated p1 and myInt" << std::endl;       // Deallocated p1 and myInt
+std::cout << pool << std::endl;                             // ObjectPool with availabilityBitmask:
+1111111111111111111111111111111111111111111111111111111111111111
+ * @endcode
+ *
+ *
+ * @tparam T Type of the blocks to allocate. Requires a default constructor.
+ * @tparam POOL_SIZE Amount of blocks to allocate
+ */
+template <class T, uint32_t POOL_SIZE>
+class ObjectPool {
+  private:
+    static constexpr uint32_t calculateAvailabilityBitmaskLenLen() {
+        if (POOL_SIZE % 8 == 0) {
+            return POOL_SIZE / 8;
+        } else {
```

Comment:
Unneccesary else, although I should probably talk to Julian about that. Goes for the rest of the file too.